src/demos/add-charts/app.js

```diff
@@ -160,31 +122,39 @@ export default class App extends Component {
     });
   }
 
+  _updateLayerSettings(settings) {
+    this.setState({settings});
+  }
+
   render() {
-    const {viewport, hoveredObject, points, settings, status, x, y} = this.state;
```

Comment:
no more deconstructions, because they make it more difficult to follow via copy/paste.